modules/javafx.graphics/src/main/java/com/sun/javafx/tk/Toolkit.java

```diff
@@ -363,76 +362,62 @@ protected Toolkit() {
 
     public abstract boolean isNestedLoopRunning();
 
-    public abstract TKStage createTKStage(Window peerWindow, boolean securityDialog, StageStyle stageStyle, boolean primary, Modality modality, TKStage owner, boolean rtl, @SuppressWarnings("removal") AccessControlContext acc);
+    public abstract TKStage createTKStage(Window peerWindow, boolean securityDialog, StageStyle stageStyle, boolean primary, Modality modality, TKStage owner, boolean rtl);
 
-    public abstract TKStage createTKPopupStage(Window peerWindow, StageStyle popupStyle, TKStage owner, @SuppressWarnings("removal") AccessControlContext acc);
-    public abstract TKStage createTKEmbeddedStage(HostInterface host, @SuppressWarnings("removal") AccessControlContext acc);
+    public abstract TKStage createTKPopupStage(Window peerWindow, StageStyle popupStyle, TKStage owner);
+    public abstract TKStage createTKEmbeddedStage(HostInterface host);
 
-    @SuppressWarnings("removal")
-    private final Map<TKPulseListener,AccessControlContext> stagePulseListeners = new WeakHashMap<>();
-    @SuppressWarnings("removal")
-    private final Map<TKPulseListener,AccessControlContext> scenePulseListeners = new WeakHashMap<>();
-    @SuppressWarnings("removal")
-    private final Map<TKPulseListener,AccessControlContext> postScenePulseListeners = new WeakHashMap<>();
-    @SuppressWarnings("removal")
-    private final Map<TKListener,AccessControlContext> toolkitListeners = new WeakHashMap<>();
+    // The following collections of listeners is weakly referenced here in order
+    // to allow garbage collection when the listeners are otherwise no longer
+    // referenced. We use a WeakHashMap with a dummy object as a value, rather
+    // than a HashSet of WeakReferences so that the entries are automatically
+    // removed after the listener is garbage collected.
+    private final Map<TKPulseListener,Object> stagePulseListeners = new WeakHashMap<>();
```

Comment:
I recommend using a set instead of a map, which is a bit less confusing:
```java
private final Set<TKPulseListener> stagePulseListeners = Collections.newSetFromMap(new WeakHashMap<>())
```